api/src/iosMain/kotlin/com/buenaflor/kflogger/util/NSThreadBasedCallerFinder.kt

```diff
@@ -0,0 +1,52 @@
+package com.buenaflor.kflogger.util
+
+import com.buenaflor.kflogger.KAbstractLogger
+import com.buenaflor.kflogger.KLogSite
+import com.buenaflor.kflogger.Klass
+import com.buenaflor.kflogger.backend.KPlatformLogCallerFinder
+import platform.Foundation.NSThread
+
+public class NSThreadBasedCallerFinder private constructor() : KPlatformLogCallerFinder() {
+  private fun findCallerNameOf(target: Klass<*>): String {
+    val name = target.kClass.qualifiedName
+    checkNotNull(name) { "Logger class has no name" }
+    val callstackSymbols = NSThread.callStackSymbols
+    check(callstackSymbols.size > 1) { "Callstack is too short" }
+    val firstIndex =
+        findIndexOfStringContainingSubstring(
+            callstackSymbols as List<String>, target.kClass.qualifiedName + ".")
+    val stackframeString = callstackSymbols[firstIndex + 1]
+    val stackFrameSignature = stackframeString.substringAfter("kfun:").substringBefore("#")
+    val lastIndexOfDot = stackFrameSignature.lastIndexOf(".")
+    check(lastIndexOfDot != -1) { "Stackframe signature does not contain a dot" }
+    if (stackFrameSignature[lastIndexOfDot + 1].isLowerCase()) {
+      // If the last character before the dot is lowercase, we assume it's a function name
+      return stackFrameSignature.substring(0, lastIndexOfDot)
+    }
+    return stackFrameSignature
```

Comment:
This returns the caller name using `NSThread.callstackSymbols`

Reply:
This code works under the assumption that KFlogger is used within the KMP modules only, so this will probably not work if KFlogger is called directly from obj-c or swift.